src/daml/Daml/Trading/Cash/CashRedemption.daml

```diff
@@ -0,0 +1,82 @@
+module Daml.Trading.Cash.CashRedemption where
+
+import Daml.Finance.Interface.Holding.Base qualified as Holding (I)
+import Daml.Trading.Cash.CashTransfer qualified as CashTransfer
+import Daml.Finance.Interface.Types.Common (AccountKey, InstrumentKey(..))
+import Daml.Finance.Interface.Holding.Fungible (ArchiveFungible(ArchiveFungible))
+import Daml.Finance.Interface.Holding.Fungible qualified as Fungible (I, Split(..), SplitResult(..))
+import Daml.Finance.Interface.Holding.Util (getAmount)
+
+template SplitRequest
+  with
+    currentOwner: Party
+    amount: Decimal
+  where
+    signatory currentOwner
+    nonconsuming choice Split : ([ContractId Fungible.I], Optional (ContractId Fungible.I))
+      with
+        holdingCid : ContractId Holding.I
+      controller currentOwner
+      do
+      -- Sanity checks
+        holding <- fetch holdingCid
+        assertMsg "Balance insufficient" (amount <= getAmount holding)
+        let
+          fungibleCid : ContractId Fungible.I = coerceContractId holdingCid
+        Fungible.SplitResult{splitCids,rest} <- exercise fungibleCid Fungible.Split with amounts = [amount]
```

Comment:
since we only have 1 amount to split we can just get the 1 single element from the `splitCids` (by using function like `head`) and return a single Contract Id instead of an array

Reply:
Done